app/main/routes_pico_api.py

```diff
@@ -77,8 +77,9 @@ def process_check_firmware(args):
 @main.route('/API/pico/getFirmware')
 @use_args(get_firmware_args, location='querystring')
 def process_get_firmware(args):
-    if args['uid'] in active_brew_sessions:
-        machine_type = active_brew_sessions[args['uid']].machine_type
+    uid = args['uid']
+    if uid in active_brew_sessions and active_brew_sessions[uid].machine_type is not None:
+        machine_type = active_brew_sessions[uid].machine_type
```

Comment:
instead of producing a server error which then translates to a Server Error 7 on the Pico device let's return "#F#" which may also produce the same experience, but at least isn't marked as a server error in a known situation where the device firmware shouldn't be loaded yet... 

will add a warning log to still have visibility to this situation